distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Dialect.enso

```diff
@@ -249,18 +239,14 @@ type Postgres_Dialect
        if the given period is supported.
     prepare_metadata_for_period : Date_Period | Time_Period -> Value_Type -> Any
     prepare_metadata_for_period self period operation_input_type =
-        case period of
-            Time_Period.Nanosecond ->
-                Error.throw (Unsupported_Database_Operation.Error "Postgres backend does not support nanosecond precision in date/time operations.")
-            _ ->
-                Date_Period_Metadata.Value period operation_input_type
+        if period == Time_Period.Nanosecond && self.name == "PostgreSQL" then Error.throw (Unsupported_Database_Operation.Error "Postgres backend does not support nanosecond precision in date/time operations.") else
+            Date_Period_Metadata.Value period operation_input_type
```

Comment:
I very much don't like this hack.

The Postgres specific implementation has `if`s for some other implementation, and it's not `if Snowflake` but `if anything else other than Postgres` which also doesn't seem right because as soon as some 3rd library started using this as well it is unclear which branch it should be landing it.

Seems OKayish to keep to get Snowflake up and running fast, but I'd really like us to plan a refactor of this in a not-too-distant future.

Reply:
Absolutely agree on this. Aiming to do Snowflake dialect in next week or so. 

The Postgres one doesn't work entirely as snowflake doesn't have same types or functions :cry: